packages/nimble-components/src/tabs/specs/scrollable-tabs.md

```diff
@@ -0,0 +1,106 @@
+
+#### TabsToolbar
+
+Only the tabs themselves will be scrollable. The scroll buttons will only surround the tabs. If there is content being displayed in a `TabsToolbar` that has been slotted, and the tabs content is wide enough to merit showing the scroll buttons, the right scroll button will sit to the left of the `TabsToolbar` (example below).
+
+![Scrollable Tabs with Toolbar](./spec-images/scrollable-tabs-toolbar.png)
+
+### Keyboard interaction
+
+#### `PageUp/PageDown`
+
+Now that the tabs will be able to be scrolled, it makes sense to add a keyboard interaction that will perform this scroll without the need of using the buttons. When a tab in the tab list has focus, pressing `PageUp/PageDown` will perform the same scroll behavior as pressing the scroll buttons _and additionally_ jump the tab focus by the same page amount (or to the first/last tab). This is needed to make the scrolling of the tabs useful for a keyboard-only user.
+
+#### Scroll buttons
+
+While there are no explicit ARIA guidelines for how dynamic buttons in something like a tablist should respond to the keyboard, the [interaction pattern for the carousel](https://www.w3.org/WAI/ARIA/apg/patterns/carousel/#keyboardinteraction) is fairly similar to this use-case, which suggests that the scroll buttons could be valid tab stops.
+
+However, as described in the ['Fundamental Keyboard Navigation Conventions'](https://www.w3.org/WAI/ARIA/apg/practices/keyboard-interface/#fundamentalkeyboardnavigationconventions) section, it states:
+
+_"A primary keyboard navigation convention common across all platforms is that the tab and shift + tab keys move focus from one UI component to another while other keys, primarily the arrow keys, move focus inside of components that include multiple focusable elements"_.
+
+The above statement would suggest that we might implement a roving tab index for the focusable elements within the `Tabs` component, whereby a user would use the arrow keys (once some element within the `Tabs` component has focus via the `Tab` key) to move between the interactive elements including the scroll buttons.
+
+**Carousel strategy**:
+
+Pros:
+
+-   Easiest to implement (i.e. it's free as it behaves this way out of the box)
+-   Not too cumbersome as one shouldn't expect a particular page to have a multitude of `Tabs` components, and additionally, the extra tabs are only if the buttons are even visible.
+
+Cons:
+
+-   After interacting with a scroll button via the keyboard a user would be required to use the mouse to select a tab that is currently in view (if a completely new set of tabs was scrolled into view).
+-   Possibly goes against conventions for composite components
+-   Possibly not in-line with other Nimble conventions such as not having the inc/dec buttons on a numeric behave as tab stops.
+
+Related:
+
+-   The [Shoelace design system](https://shoelace.style/components/tab-group#scrolling-tabs) has an example of this approach
+
+**Roving tab strategy**:
```

Comment:
Could you give an example or otherwise describe the roving tab strategy?

Reply:
For starters I misnamed this section, which I will correct. It should be "Roving tab _index_ strategy".

If implemented for the `Tabs` it would simply mean that pressing `<Tab>` when the next tabbable component on the page is the `Tabs` component would focus either the first or most recently focused `Tab`. Then to move focus to the other interactive components you would use the arrow keys. I think the main distinction between what I am proposing in this HLD and a typical implementation of the roving tab index, is that the scroll buttons would still never be given a `tabindex` of 0 (which would allow them to receive tab focus if a user tabbed again to the `Tabs` component). They would only be reachable via the arrow keys.

An example in Nimble that has implemented this strategy is the `Toolbar`.